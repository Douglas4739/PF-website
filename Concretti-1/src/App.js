

// TODO
// Yefat selects the color of the products that need to be manufactured.
// Selects the products that need to be built in that color.
// Then the app spits out the total material cost of those products (use x amount of white Quickcrete, x amount of pigment, ect).

function App() {
  return (
    <>
    </>
  );
}

export default App;
