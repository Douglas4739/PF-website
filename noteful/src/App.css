.App {
    text-align: center;
    align-items: center;
    justify-content: center;
    background-color: #ffffff;
  }
  
  .App-logo {
    height: 40vmin;
    pointer-events: none;
  }
  
  @media (prefers-reduced-motion: no-preference) {
    .App-logo {
      animation: App-logo-spin infinite 20s linear;
    }
  }
  
  .App-header {
    background-color: #c2c2c2;
    min-height: 100vh;
    display: flex;
    flex-direction: row;
    align-items: center !important;
    justify-content: center !important;
    font-size: calc(10px + 2vmin);
    color: rgb(145, 5, 5);
  }
  
  .App-link {
    color: #ff0000;
  }
  
  @keyframes App-logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }