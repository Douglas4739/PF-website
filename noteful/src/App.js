import React from 'react';
import {withRouter, Route, Switch} from 'react-router-dom';
import Header from './Main/Header';
import ErrorPage from './Main/ErrorPage';
import ErrorFolder from './Folders/ErrorFolder';
import RuntimeError from './RuntimeError';
import SideBar from './Main/SideBar';
import MainPage from './Main/MainPage';
import NotePage from './Notes/NotePage';
import AddFolder from './Folders/AddFolder';
import NoteContext from './Notes/NoteContext';
import * as api from './api';
import './App.css';

class App extends React.Component {
  state = { folders: [], notes: [] };

  addFolder = (folderName) => {
    const newState = {...this.state};
    api.addFolder(folderName)
    .then(folderObj => {
      folderObj.id = folderObj.id.toString();
      newState.folders.push(folderObj);
      this.setState({...newState});
      this.props.history.push(`/folders/${folderObj.id}`);
    });
  };

  editFolder = (folderName, folderID) => {
    const newState = {...this.state};
    const folders = newState.folders.map(folder => {
      return (
        (folder.id.toString() === folderID)
          ? Object.assign(folder, {name: folderName})
          : folder
      );
    });
    this.setState({...newState, folders});
    this.props.history.push(`/folders/${folderID}`);
    api.editFolder(folderName, folderID);
  };

  deleteFolder = (folderID) => {
    const folders = [...this.state.folders].filter(folder => folder.id !== folderID);
    const notes = [...this.state.notes].filter(note => note.folderId !== folderID);
    this.setState( {folders, notes} );
    this.props.history.push('/');
    api.deleteFolder(folderID);
  }

  addNote = (note) => {
    const newState = {...this.state};
    Object.assign(note,
      {modified: (new Date()).toISOString()}
    );
    api.addNote(note)
    .then(serverNote => {
      newState.notes.push(serverNote);
      this.setState({...newState});
      this.props.history.push(`/folders/${note.folderId}`);
    });
  };

  editNote = (editedNote) => {
    const newState = {...this.state};
    const newNote = newState.notes.find(note => note.id === editedNote.id);
    Object.assign(newNote,
      editedNote,
      {modified: new Date().toISOString()}
    );
    const notes = newState.notes.map(note => {
      return (
        (note.id === editedNote.id)
          ? newNote
          : note
      );
    });
    this.setState({...newState, notes});
    this.props.history.push(`/folders/${newNote.folderId}`);
    api.editNote(newNote);
  };

  deleteNote = (noteID) => {
    const newState = {...this.state};
    const deletedNote = newState.notes.find(note => note.id === noteID);
    const notes = newState.notes.filter(note => note.id !== noteID);
    this.setState({...newState, notes});
    this.props.history.push(`/folders/${deletedNote.folderId}`);
    api.deleteNote(noteID);
  }

  render() {

    return (
      <main className='App'>
        <RuntimeError show='header'>
          <Header />
        </RuntimeError>
        <div id='main-section'>
          <NoteContext.Provider value={{
            deleteNote: this.deleteNote,
            addNote: this.addNote,
            editNote: this.editNote
          }}>
            <Switch>
              <Route exact path='/folders/:folder/notes/:note' render={() =>
                <RuntimeError show='header'>
                  <SideBar notes={this.state.notes} />
                  <NotePage folders={this.state.folders} notes={this.state.notes} />
                </RuntimeError>
              } />
              <Route exact path={['/', '/folders/:folder/']} render={() => 
                <RuntimeError show='page'>
                  <SideBar folders={this.state.folders} />
                  <MainPage state={this.state} deleteFolder={this.deleteFolder} />
                </RuntimeError>
              } />
              <Route exact path={['/new-folder', '/editfolder/:folder']} render={() => 
                <RuntimeError show='new folder form'>
                  <AddFolder
                    addFolder={this.addFolder}
                    editFolder={this.editFolder}
                    deleteFolder={this.deleteFolder}
                    folders={this.state.folders}
                  />
                </RuntimeError>
              }/>
              <Route exact path='/folders/:folder/new-note' render={() =>
                <RuntimeError show='note page'>
                  <SideBar notes={this.state.notes} />
                  <NotePage folders={this.state.folders}/>
                </RuntimeError>
              } />
              <Route path='/choose-folder' render={() => 
                <RuntimeError show='note page'>
                  <ErrorFolder />
                </RuntimeError>
              } />
              <Route render={() => 
                <ErrorPage show='page' /> 
              }/>
            </Switch>
          </NoteContext.Provider>
        </div>
      </main>
    );
  };

  componentDidMount() {
    api.getData()
    .then(data => {
      const [folders, notes] = data;
      this.setState({folders, notes});
    })
    .catch(error => {
      console.log(`Could not fetch data. Error: ${error.message}`);
    });
  };
};

export default withRouter(App);