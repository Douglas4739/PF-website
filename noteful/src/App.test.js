import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';

describe('The App', () => {
  it('renders without fail.', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('shows no breaking changes.', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});