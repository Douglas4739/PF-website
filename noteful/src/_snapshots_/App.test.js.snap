// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App shows no breaking changes. 1`] = `
<main
  className="App"
>
  <div
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      NOTEFUL
    </a>
  </div>
  <div
    id="main-section"
  >
    <div
      className="sidebar"
    >
      <div
        className="side-previews"
      />
      <div
        className="button"
        style={
          Object {
            "height": "initial",
          }
        }
      >
        <a
          href="/new-folder"
          id="add-folder"
          onClick={[Function]}
        >
          Add Folder
        </a>
      </div>
    </div>
    <div
      className="note-list"
    >
      <h3 />
      <div
        className="active-folder-options"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <button
          id="active-edit"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          id="active-delete"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
      <a
        href="/choose-folder"
        id="add-note"
        onClick={[Function]}
      >
        Add Note
      </a>
    </div>
  </div>
</main>
`;